rust-install/src/component/components.rs

```diff
@@ -36,22 +40,23 @@ impl<'a> ChangeSet<'a> {
 	}
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct Components {
 	prefix: InstallPrefix,
 }
 
 impl Components {
-	pub fn new(prefix: InstallPrefix) -> Option<Self> {
-		if utils::is_file(prefix.manifest_file(COMPONENTS_FILE)) {
-			Some(Components { prefix: prefix })
-		} else {
-			None
-		}
-	}
-	pub fn init(prefix: InstallPrefix) -> Result<Self> {
-		try!(utils::write_file("components", &prefix.manifest_file(COMPONENTS_FILE), ""));
-		Ok(Components { prefix: prefix })
```

Comment:
Could you explain why `init` was removed? The `new` -> `open` and Option -> Result changes make sense, although the other "InstallPrefix wrappers" for lack of a better term should probably be refactored in the same way (Configuration and Manifestation). `init` was required for initializing an fresh install.